packages/core/src/workflows/machine.ts

```diff
@@ -79,7 +79,7 @@ export class Machine<
     executionSpan?: Span;
     name: string;
     runId: string;
-    steps: Record<string, IAction<any, any, any, any>>;
+    steps: Record<string, StepAction<any, any, any, any>>;
```

Comment:
are these compatible? looks like they have different signatures

```ts
  id: TId;
  description?: string;
  inputSchema?: TSchemaIn;
  outputSchema?: TSchemaOut;
```
vs
```ts
  mastra?: Mastra;
  payload?: TSchemaIn extends z.ZodSchema ? Partial<z.infer<TSchemaIn>> : unknown;
  retryConfig?: RetryConfig;
```
Or was it a mistake that workflow steps were typed as IAction?

Reply:
No looks like they're not compatible https://vercel.com/kepler-ab1bf5ab/travel-ai/CPtSv6Z7XwfMy8XUWJKDDE1nAgRH?filter=errors

Reply:
oh that error is on main 🤔 think it's because of this https://github.com/mastra-ai/mastra/pull/2863

Reply:
steps should be TSteps

Reply:
TSteps is an array and this is an object
I guess you could do `  #steps: Record<string, TSteps[0]> = {};` ?

Reply:
We have another type but i'll fix it after this as travel ai is indeed not related